nvme-print-json.c

```diff
@@ -2197,26 +2205,25 @@ static void json_phy_rx_eom_descs(struct nvme_phy_rx_eom_log *log,
 
 		if (desc->edlen == 0)
 			continue;
-		else {
-			/* Hex dump Vendor Specific Eye Data*/
-			vsdata = (unsigned char *)malloc(desc->edlen);
-			vsdataoffset = (desc->nrows * desc->ncols) +
-								sizeof(struct nvme_eom_lane_desc);
-			vsdata = (unsigned char *)((unsigned char *)desc + vsdataoffset);
-			char *hexstr = malloc(desc->edlen * 3 + 1); // 2 hex chars + space per byte
 
-			if (!hexstr)
-				return;
+		/* Hex dump Vendor Specific Eye Data*/
+		vsdataoffset = (le16_to_cpu(desc->nrows) * le16_to_cpu(desc->ncols)) +
+						sizeof(struct nvme_eom_lane_desc);
+		vsdata = (unsigned char *)((unsigned char *)desc + vsdataoffset);
+		// 2 hex chars + space per byte
+		_cleanup_free_ char *hexstr = malloc(le16_to_cpu(desc->edlen) * 3 + 1);
```

Comment:
Please introduce local variable for `nrows` `ncols` and `edlen` so you only need to to the cpu conversion once.

Also while at it please use consistent comment style.